convert2rhel/actions/pre_ponr_changes/backup_system.py

```diff
@@ -63,3 +79,81 @@ def run(self):
 
         repo.backup_yum_repos()
         repo.backup_varsdir()
+
+
+class BackupPackageFiles(actions.Action):
+    id = "BACKUP_PACKAGE_FILES"
+
+    def run(self):
+        """Backup changed package files"""
+        super(BackupPackageFiles, self).run()
+
+        loggerinst.task("Prepare: Backup package files")
+
+        package_files_changes = self._get_changed_package_files()
+
+        for file in package_files_changes:
+            if file["status"] == "missing":
+                pass  # no action needed now
+            elif "5" in file["status"]:
+                # If the MD5 checksum differs, the content of the file differs
+                restorable_file = backup.NewRestorableFile(file["path"])
+                backup.backup_control.push(restorable_file)
+
+    def _get_changed_package_files(self):
+        """Get the output from rpm -Va command from during resolving system info
+        to get changes made to package files.
+        Return them as a list of dict, for example:
+        [{"status":"S5T", "file_type":"c", "path":"/etc/yum.repos.d/CentOS-Linux-AppStream.repo"}]
+        """
+        path = os.path.join(LOG_DIR, PRE_RPM_VA_LOG_FILENAME)
+        with open(path, "r") as f:
+            output = f.read()
+
+        output = output.strip()
+        lines = output.split("\n")
+        data = []
+
+        for line in lines:
+            parsed_line = self._parse_line(line.strip())
+            if parsed_line["path"] and parsed_line["status"]:
+                data.append(parsed_line)
+
+        return data
+
+    def _parse_line(self, line):
+        """Return {"status":"S5T", "file_type":"c", "path":"/etc/yum.repos.d/CentOS-Linux-AppStream.repo"}"""
+        match = re.match(RPM_VA_REGEX, line)
+
+        if not match:  # line not matching the regex
+            if line.strip() != "":
+                # Line is not empty string
+                loggerinst.debug("Skipping invalid output %s" % line)
+            return {"status": None, "file_type": None, "path": None}
+
+        line = line.split()
+
+        # Replace the . (success) and ? (test could not be performed) symbols
+        status = line[0].replace(".", "").replace("?", "")
+
+        if len(line) == 2:
+            # File type undefined
+            file_type = None
+            path = line[1]
+
+        else:
+            file_type = line[1]
+            path = line[2]
+
+        return {"status": status, "file_type": file_type, "path": path}
+
+    @staticmethod
+    def rollback_files():
```

Comment:
Is this tested anywhere?

Reply:
@SpyTec It's not used anywhere, probably some leftover from previous implementation without the `NewRestorableFile`. But then it seems to me that I don't handle the missing files - not removing them. I'll take a look if I'm right